app/src/main/java/ch/admin/bag/dp3t/storage/SecureStorage.java

```diff
@@ -285,4 +289,13 @@ public WhatToDoPositiveTestTextsModel getWhatToDoPositiveTestTexts(String langua
 		return map.get(language);
 	}
 
+	public void setTestLocations(Map<String, String> testLocations) {
+		prefs.edit().putString(KEY_TEST_LOCATIONS, gson.toJson(testLocations)).apply();
+	}
+
+	public Map<String, String> getTestLocations() {
+		Type type = new TypeToken<HashMap<String, String>>() { }.getType();
+		return gson.fromJson(prefs.getString(KEY_TEST_LOCATIONS, "{}"), type);
```

Comment:
We should consider adding a default canton list to the app, in case the user visits the page before a successfull config request.